internal/resources/tanzupackageinstall/package_install_datasource.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package tanzupackageinstall
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/tanzupackageinstall/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/tanzupackageinstall/spec"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/tanzupackageinstall/status"
+)
+
+func DataSourcePackageInstall() *schema.Resource {
+	return &schema.Resource{
+		Schema: getDataSourceSchema(),
+		ReadContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			return dataPackageInstallRead(context.WithValue(ctx, contextMethodKey{}, DataSourceRead), d, m)
+		},
+	}
+}
+
+func dataPackageInstallRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	packageInstallName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package install name")
+	}
+
+	packageInstallNamespace, ok := d.Get(NamespaceKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package install name")
```

Comment:
"Unable to read package install namespace" ??

Reply:
@Shubbhang351 ?